packages/css/src/index.js

```diff
@@ -125,28 +125,53 @@ const transforms = [
 export const responsive = styles => theme => {
   const next = {}
   const breakpoints = get(theme, 'breakpoints', defaultBreakpoints)
-  const mediaQueries = [
-    null,
-    ...breakpoints.map(n => `@media screen and (min-width: ${n})`),
-  ]
+  let mediaQueries
+  if(Array.isArray(breakpoints)) {
+    mediaQueries = [null, ...breakpoints.map((n) => `@media screen and (min-width: ${n})`)]
+  }
+  else {
+    mediaQueries = {}
+    for (const key in breakpoints) {
+      mediaQueries[key] = `@media screen and (min-width: ${breakpoints[key]})`
+    }
+  }
 
   for (const key in styles) {
     const value = styles[key]
     if (value == null) continue
-    if (!Array.isArray(value)) {
-      next[key] = value
-      continue
-    }
-    for (let i = 0; i < value.length; i++) {
-      const media = mediaQueries[i]
-      if (value[i] == null) continue
-      if (!media) {
-        next[key] = value[i]
+    if(typeof value === 'object') {
+      if (Array.isArray(value)) {
+        for (const i in value) {
+          const media = mediaQueries[i]
+          if (value[i] == null) continue
+          if (!media) {
+            next[key] = value[i]
+            continue
+          }
+          next[media] = next[media] || {}
+          next[media][key] = value[i]
+        }
+        continue
+      }
+      else {
+        for (const k in value) {
+          const media = mediaQueries[k]
+          if (value[k] == null) continue
+          if(k === '_' && value._) {
+            next[key] = value._
```

Comment:
Why are you checking specifically for `_` here? 

Is this the object convention to specify the "mobile first" property? (equivalent to the first value in an array prop)

Would the following code work:
```
css({
    color: 'primary',
    fontSize: { default: 2, small: 3, medium: 4 }
})
```

Reply:
Good point!

` _` is probably not the default "mobile first" property. I'll have to check if I can rewrite that part.

The whole `@styled-system/css` feels a bit outdated tho..

Reply:
Cool, let me know when you do and I can re-review 👍